liblocation/src/main/java/com/mapbox/android/core/location/AndroidLocationEngineImpl.java

```diff
@@ -141,13 +142,23 @@ private static int priorityToPowerRequirement(int priority) {
   @VisibleForTesting
   static final class AndroidLocationEngineCallbackTransport implements LocationListener {
     private final LocationEngineCallback<LocationEngineResult> callback;
+    private final LocationEngineRequest request;
+    private long lastUpdateTime = 0;
 
-    AndroidLocationEngineCallbackTransport(LocationEngineCallback<LocationEngineResult> callback) {
+    AndroidLocationEngineCallbackTransport(@NonNull LocationEngineRequest request,
+                                           LocationEngineCallback<LocationEngineResult> callback) {
+      this.request = request;
       this.callback = callback;
     }
 
     @Override
     public void onLocationChanged(Location location) {
+      long currentTime = System.currentTimeMillis();
```

Comment:
In testcase it would always be 0 and fail our test, so let's keep it.

Reply:
What do you mean? should our tests drive internal implementation details? the test is failing because `SystemClock` is android framework class and we're using `unitTests.returnDefaultValues = true` by default. That said `System.currentTimeMillis()` is not safe for measuring time spans, so let's try to find workaround.

Reply:
In Unit test `SystemClock.elapsedRealtime()` will always be 0, but `System.currentTimeMillis()` can get the real time.